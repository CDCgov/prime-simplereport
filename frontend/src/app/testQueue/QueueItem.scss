.prime-card-container {
  .card-header {
    display: flex;
    justify-content: flex-start;
    align-items: center;

    .card-name {
      font-size: 22px;
      font-weight: 700;
      color: #005ea2;
      cursor: pointer;
    }

    .card-dob {
      margin-left: 24px;
      font-size: 16px;
      font-weight: 400;
      color: #71767a;

      .card-date {
        font-weight: 700;
      }
    }
  }

  .card-test-input {
    height: 34px;
    margin-right: 12px;
    max-width: 130px;
    animation: 0.25s fadeIn;
    animation-fill-mode: both;

    @keyframes fadeIn {
      0% {
        opacity: 0;
      }

      100% {
        visibility: visible;
        opacity: 1;
      }
    }

    &__error {
      border-color: #d54309;
    }
  }

  .card-dropdown {
    select {
      height: 34px;
      padding-top: 4px;
      padding-bottom: 4px;
    }
  }

  input[type="date"]::-webkit-calendar-picker-indicator {
    margin-left: 0;
  }

  .check-box-container {
    display: flex;
    align-items: center;
    height: 34px;
  }

  .test-questionnaire-btn {
    font-weight: 700 !important;
    margin-bottom: 1em;
  }

  .flex-col-container {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    flex-wrap: wrap;
  }

  .test-date-time-container {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: wrap;
    margin-top: 8px;
  }

  .queue-item-error-message {
    color: #d54309;
    display: block;
  }
}

.symptom-onset-date {
  height: 40px;
  width: 200px;
  margin-top: 8px;
}
