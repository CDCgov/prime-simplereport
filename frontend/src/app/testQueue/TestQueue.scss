@use "../../scss/utilities";

.grid-container.queue-container-wide {
  max-width: utilities.$site-max-width;
}

.prime-test-result {
  .usa-radio {
    white-space: nowrap;
  }
}
