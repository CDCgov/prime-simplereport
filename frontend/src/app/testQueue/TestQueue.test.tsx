import {
  render,
  screen,
  waitForElementToBeRemoved,
  within,
} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MockedProvider } from "@apollo/client/testing";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore, { MockStoreEnhanced } from "redux-mock-store";

import TestQueue, { queueQuery } from "./TestQueue";
import { REMOVE_PATIENT_FROM_QUEUE } from "./QueueItem";
import { QUERY_PATIENT } from "./addToQueue/AddToQueueSearch";

jest.mock("@microsoft/applicationinsights-react-js", () => {
  return {
    useAppInsightsContext: jest.fn(),
    useTrackEvent: () => jest.fn(),
  };
});

describe("TestQueue", () => {
  let store: MockStoreEnhanced<unknown, {}>;
  const mockStore = configureStore([]);

  beforeEach(() => {
    store = mockStore({
      organization: {
        name: "Organization Name",
      },
      facilities: [
        {
          id: "a1",
          name: "Fake Facility",
        },
      ],
    });
  });

  it("should render the test queue", async () => {
    jest
      .useFakeTimers("modern")
      .setSystemTime(new Date("2021-08-01 08:20").getTime());
    const { container } = render(
      <MemoryRouter>
        <MockedProvider mocks={mocks}>
          <Provider store={store}>
            <TestQueue activeFacilityId="a1" />
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );
    await screen.findByLabelText("Search");
    expect(await screen.findByText("Doe, John A")).toBeInTheDocument();
    expect(await screen.findByText("Smith, Jane")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should remove items queue using the transition group", async () => {
    render(
      <MemoryRouter>
        <MockedProvider mocks={mocks}>
          <Provider store={store}>
            <TestQueue activeFacilityId="a1" />
          </Provider>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(await screen.findByText("Doe, John A")).toBeInTheDocument();
    const removeButton = (await screen.findAllByLabelText("Close"))[0];
    userEvent.click(removeButton);
    const confirmButton = await screen.findByText("Yes", { exact: false });
    userEvent.click(confirmButton);
    expect(
      screen.getByText("Submitting test data for Doe, John A...")
    ).toBeInTheDocument();
    await waitForElementToBeRemoved(
      () => screen.queryByText("Submitting test data for Doe, John A..."),
      { timeout: 10000 }
    );
    expect(screen.queryByText("Doe, John A")).not.toBeInTheDocument();
  });

  describe("clicking on test questionnaire", () => {
    beforeEach(async () => {
      await render(
        <MemoryRouter>
          <MockedProvider mocks={mocks}>
            <Provider store={store}>
              <TestQueue activeFacilityId="a1" />
            </Provider>
          </MockedProvider>
        </MemoryRouter>
      );

      await screen.findByLabelText("Search");
      expect(await screen.findByText("Doe, John A")).toBeInTheDocument();
      expect(await screen.findByText("Smith, Jane")).toBeInTheDocument();

      userEvent.click(screen.getAllByText("Test questionnaire")[0]);
    });

    it("should open test questionnaire and display emails and phone numbers correctly", () => {
      const modal = screen.getByRole("dialog");

      expect(within(modal).getByText("Test questionnaire")).toBeInTheDocument();
      expect(
        within(modal).getByText(
          "Would you like to receive a copy of your results via text message?"
        )
      ).toBeInTheDocument();
      expect(
        within(modal).getByText(
          "Would you like to receive a copy of your results via email?"
        )
      ).toBeInTheDocument();
      expect(within(modal).getByText("Doe@legacy.com")).toBeInTheDocument();
      expect(within(modal).getByText("John@legacy.com")).toBeInTheDocument();
      expect(within(modal).getByText("8178675309")).toBeInTheDocument();
      expect(within(modal).getByText("8178675911")).toBeInTheDocument();
    });
  });

  describe("device selection", () => {
    it("should select another swab type for the selected device if configured swab not available", async () => {
      const deviceSpecimenNotAvailableResult = { ...result };

      // Default device specimen for facility is not a valid type in the `deviceSpecimenTypes` array
      // returned by GraphQL. The component should select another of the facility's devices to
      // populate the dropdown initially
      deviceSpecimenNotAvailableResult.data.organization.testingFacility[0].defaultDeviceSpecimen =
        "c0c7b042-9a4f-47cd-b280-46c0daa51c86";
      deviceSpecimenNotAvailableResult.data.deviceSpecimenTypes.pop();

      const mock = {
        request: {
          query: queueQuery,
          variables: {
            facilityId: "a1",
          },
        },
        result: deviceSpecimenNotAvailableResult,
      };

      render(
        <MemoryRouter>
          <MockedProvider mocks={[mock]}>
            <Provider store={store}>
              <TestQueue activeFacilityId="a1" />
            </Provider>
          </MockedProvider>
        </MemoryRouter>
      );

      expect(
        ((await screen.findAllByText("Quidel Sofia 2"))[0] as HTMLOptionElement)
          .selected
      ).toBeTruthy();

      expect(
        ((
          await screen.findAllByText("Nasopharyngeal swab")
        )[0] as HTMLOptionElement).selected
      ).toBeTruthy();
    });

    it("should select another device if the configured device has no associated swab types", async () => {
      const deviceNotAvailableResult = { ...result };

      // The device configured as the facility default has no associated swab types - dropdown
      // should be populated with another valid device+swab type combo
      deviceNotAvailableResult.data.queue[0].deviceType = {
        internalId: internalId,
        model: "lumira",
        name: "LumiraDx",
        testLength: 15,
        __typename: "DeviceType",
      };

      deviceNotAvailableResult.data.deviceSpecimenTypes.shift();

      const mock = {
        request: {
          query: queueQuery,
          variables: {
            facilityId: "a1",
          },
        },
        result: deviceNotAvailableResult,
      };

      render(
        <MemoryRouter>
          <MockedProvider mocks={[mock]}>
            <Provider store={store}>
              <TestQueue activeFacilityId="a1" />
            </Provider>
          </MockedProvider>
        </MemoryRouter>
      );

      // The facility's default device is "LumiraDx", but that device has no associated swabs
      expect(
        ((await screen.findAllByText("Quidel Sofia 2"))[0] as HTMLOptionElement)
          .selected
      ).toBeTruthy();
    });
  });
});

const internalId = "f5c7658d-a0d5-4ec5-a1c9-eafc85fe7554";

const symptoms = JSON.stringify({
  "64531003": "false",
  "103001002": "false",
  "84229001": "false",
  "68235000": "false",
  "426000000": "false",
  "49727002": "false",
  "68962001": "false",
  "422587007": "false",
  "267036007": "false",
  "62315008": "false",
  "43724002": "false",
  "36955009": "false",
  "44169009": "false",
  "422400008": "false",
  "230145002": "false",
  "25064002": "false",
  "162397003": "false",
});

const createPatient = ({
  first,
  middle,
  last,
  birthDate,
  resultId,
}: {
  first: string;
  middle: string | null;
  last: string;
  birthDate: string;
  resultId: string;
}) => ({
  internalId: resultId,
  pregnancy: null,
  dateAdded: "2021-02-11 20:53:51.337",
  symptoms,
  symptomOnset: null,
  noSymptoms: false,
  patient: {
    internalId: resultId,
    telephone: "1234567890",
    birthDate,
    firstName: first,
    middleName: middle,
    lastName: last,
    email: `${middle}@legacy.com`,
    emails: [`${first}@legacy.com`, `${last}@legacy.com`],
    phoneNumbers: [
      {
        number: "8178675309",
        type: "MOBILE",
      },
      {
        number: "8178675911",
        type: "MOBILE",
      },
    ],
    gender: "female",
    testResultDelivery: "",
    preferredLanguage: [],
    __typename: "Patient",
  },
  deviceType: {
    internalId: "0f3d7426-3476-4800-97e7-3de8a93b090c",
    model: "quidel",
    name: "Quidel Sofia 2",
    testLength: 10,
    __typename: "DeviceType",
  },
  deviceSpecimenType: {
    internalId,
  },
  result: "",
  dateTested: null,
  patientLink: {
    internalId: "7df95d14-c9ca-406e-bed7-85da05d5eea1",
    __typename: "PatientLink",
  },
  __typename: "TestOrder",
});

// Mock data taken from network request tab in Chrome devtools
const result = {
  data: {
    queue: [
      createPatient({
        first: "John",
        middle: "A",
        last: "Doe",
        birthDate: "1996-06-19",
        resultId: "abc",
      }),
      createPatient({
        first: "Jane",
        middle: null,
        last: "Smith",
        birthDate: "2021-02-01",
        resultId: "def",
      }),
    ],
    organization: {
      testingFacility: [
        {
          id: "a1",
          deviceTypes: [
            {
              internalId,
              testLength: 15,
              model: "lumira",
              name: "LumiraDX",
              __typename: "DeviceType",
            },
            {
              internalId: "0f3d7426-3476-4800-97e7-3de8a93b090c",
              testLength: 15,
              model: "quidel",
              name: "Quidel Sofia 2",
              __typename: "DeviceType",
            },
          ],
          defaultDeviceSpecimen: internalId,
          __typename: "Facility",
        },
      ],
      __typename: "Organization",
    },
    deviceSpecimenTypes: [
      {
        internalId: "3d5c0c67-cddb-4344-8037-18008d6fe809",
        deviceType: {
          internalId: internalId,
          model: "lumira",
          name: "LumiraDx",
          testLength: 15,
          __typename: "DeviceType",
        },
        specimenType: {
          internalId: "6aa957dc-add2-4938-8788-935aec3276d4",
          name: "Swab of internal nose",
          __typename: "SpecimenType",
        },
        __typename: "DeviceSpecimenType",
      },
      {
        internalId: "f8b9d9d6-c318-4c54-a516-76f0d9a25d32",
        deviceType: {
          internalId: "0f3d7426-3476-4800-97e7-3de8a93b090c",
          model: "quidel",
          name: "Quidel Sofia 2",
          testLength: 10,
          __typename: "DeviceType",
        },
        specimenType: {
          internalId: "6e4ccb35-d177-4ea0-9226-653358f1e081",
          name: "Nasopharyngeal swab",
          __typename: "SpecimenType",
        },
        __typename: "DeviceSpecimenType",
      },
      {
        internalId: "c0c7b042-9a4f-47cd-b280-46c0daa51c86",
        deviceType: {
          internalId: "0f3d7426-3476-4800-97e7-3de8a93b090c",
          model: "quidel",
          name: "Quidel Sofia 2",
          testLength: 10,
          __typename: "DeviceType",
        },
        specimenType: {
          internalId: "6aa957dc-add2-4938-8788-935aec3276d4",
          name: "Swab of internal nose",
          __typename: "SpecimenType",
        },
        __typename: "DeviceSpecimenType",
      },
    ],
  },
};

const mocks = [
  {
    request: {
      query: queueQuery,
      variables: {
        facilityId: "a1",
      },
    },
    result,
  },
  {
    request: {
      query: QUERY_PATIENT,
      variables: {
        facilityId: "a1",
        namePrefixMatch: "",
      },
    },
    result: {},
  },
  {
    request: {
      query: REMOVE_PATIENT_FROM_QUEUE,
      variables: {
        patientId: "abc",
      },
    },
    result: {
      data: { removePatientFromQueue: null },
    },
  },
  {
    request: {
      query: queueQuery,
      variables: {
        facilityId: "a1",
      },
    },
    result: {
      data: {
        ...result.data,
        queue: result.data.queue.slice(1),
      },
    },
  },
];
