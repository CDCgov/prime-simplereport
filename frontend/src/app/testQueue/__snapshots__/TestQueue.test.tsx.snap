// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestQueue should render the test queue 1`] = `
<div>
  <main
    class="prime-home"
    style="padding-top: 80px;"
  >
    <div
      class="grid-container"
    >
      <div
        class="position-relative"
      >
        <div
          class="prime-search-container"
        >
          <form
            class="usa-search usa-search--small prime-search-input display-inline-block"
            role="search"
          >
            <label
              class="usa-sr-only"
              for="search-field-small"
            >
              Search
            </label>
            <div>
              <input
                autocomplete="off"
                class="usa-input"
                id="search-field-small"
                name="search"
                placeholder="Search for a person to start their test"
                type="search"
                value=""
              />
              <button
                class="usa-button"
                disabled=""
                type="submit"
              >
                <span
                  class="usa-sr-only"
                >
                  Search
                </span>
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="position-relative grid-container prime-container prime-queue-item card-container"
        data-testid="test-card-abc"
      >
        <div
          aria-hidden="true"
          class="sr-queue-item-submit-loader z-top position-absolute height-full width-full text-center radius-lg"
        >
          <h4
            class="margin-top-6 margin-bottom-5"
          >
            Submitting test data for 
            Doe, John A
            ...
          </h4>
          <img
            alt="submitting"
            class="square-5"
            src="loader.svg"
          />
        </div>
        <div
          class="prime-card-container"
        >
          <button
            aria-label="Close"
            class="prime-close-button"
          >
            <span
              class="fa-layers"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle fa-w-16 fa-inverse fa-2x "
                data-icon="circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times-circle fa-w-16 fa-2x "
                data-icon="times-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <div
            class="grid-row"
          >
            <div
              class="tablet:grid-col-9"
            >
              <div
                class="grid-row prime-test-name usa-card__header"
                id="patient-name-header"
              >
                <div
                  class="card-header"
                >
                  <div
                    class="card-name"
                  >
                    Doe, John A
                  </div>
                  <div
                    class="card-dob"
                  >
                    Date of birth:
                    <span
                      class="card-date"
                    >
                       
                      06/19/1996
                    </span>
                  </div>
                </div>
                <button
                  class="timer-button timer-reset"
                  data-testid="timer"
                >
                  <span>
                    10:00
                  </span>
                   
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-stopwatch fa-w-14 "
                    data-icon="stopwatch"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M432 304c0 114.9-93.1 208-208 208S16 418.9 16 304c0-104 76.3-190.2 176-205.5V64h-28c-6.6 0-12-5.4-12-12V12c0-6.6 5.4-12 12-12h120c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-28v34.5c37.5 5.8 71.7 21.6 99.7 44.6l27.5-27.5c4.7-4.7 12.3-4.7 17 0l28.3 28.3c4.7 4.7 4.7 12.3 0 17l-29.4 29.4-.6.6C419.7 223.3 432 262.2 432 304zm-176 36V188.5c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12V340c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="margin-top-2 margin-left-2 margin-bottom-2"
              >
                <div
                  class="grid-row"
                >
                  <div
                    class="grid-col-4 flex-col-container padding-right-2"
                  >
                    <button
                      class="usa-button usa-button--unstyled test-questionnaire-btn"
                      type="button"
                    >
                      Test questionnaire
                    </button>
                    <div
                      class="margin-bottom-1"
                    >
                      <span
                        class="usa-tag"
                      >
                        PENDING
                      </span>
                    </div>
                  </div>
                  <div
                    class="flex-col-container"
                  >
                    <div
                      class=""
                    >
                      Test date and time
                    </div>
                    <div
                      class="test-date-time-container"
                    >
                      <input
                        class="card-test-input"
                        data-testid="test-date"
                        hidden=""
                        id="test-date"
                        max="2021-08-01"
                        min="2020-01-01"
                        name="test-date"
                        type="date"
                        value="2021-08-01"
                      />
                      <input
                        class="card-test-input"
                        data-testid="test-time"
                        hidden=""
                        name="test-time"
                        step="60"
                        type="time"
                        value="08:20"
                      />
                      <div
                        class="check-box-container"
                      >
                        <div
                          class="usa-checkbox"
                        >
                          <input
                            checked=""
                            class="usa-checkbox__input margin"
                            id="current-date-check-abc"
                            type="checkbox"
                            value="true"
                          />
                          <label
                            class="usa-checkbox__label margin-0 margin-right-05em"
                            for="current-date-check-abc"
                          >
                            Current date/time
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="queue-item__form prime-ul grid-row"
                >
                  <div
                    class="prime-li flex-align-self-end tablet:grid-col-4 padding-right-2"
                  >
                    <div
                      class="usa-form-group prime-dropdown  card-dropdown"
                    >
                      <label
                        class="usa-label"
                        for="1"
                      >
                        Device
                      </label>
                      <select
                        aria-required="false"
                        class="usa-select"
                        id="1"
                        name="testDevice"
                      >
                        <option
                          value="f5c7658d-a0d5-4ec5-a1c9-eafc85fe7554"
                        >
                          LumiraDx
                        </option>
                        <option
                          value="0f3d7426-3476-4800-97e7-3de8a93b090c"
                        >
                          Quidel Sofia 2
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    class="prime-li flex-align-self-end tablet:grid-col-5 padding-right-2"
                  >
                    <div
                      class="usa-form-group prime-dropdown  card-dropdown"
                    >
                      <label
                        class="usa-label"
                        for="2"
                      >
                        Swab type
                      </label>
                      <select
                        aria-required="false"
                        class="usa-select"
                        id="2"
                        name="swabType"
                      >
                        <option
                          value="6e4ccb35-d177-4ea0-9226-653358f1e081"
                        >
                          Nasopharyngeal swab
                        </option>
                        <option
                          value="6aa957dc-add2-4938-8788-935aec3276d4"
                        >
                          Swab of internal nose
                        </option>
                      </select>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="tablet:grid-col-3 prime-test-result"
            >
              <form
                class="usa-form"
              >
                <h4
                  class="prime-radio__title"
                >
                  SARS-CoV-2 results
                </h4>
                <div
                  class="usa-form-group prime-radio__group"
                >
                  <fieldset
                    class="usa-fieldset prime-radios"
                  >
                    <legend
                      class="usa-legend usa-sr-only"
                    >
                      Test result
                    </legend>
                    <div
                      class=""
                    >
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="3-val-POSITIVE"
                          name="covid-test-result-abc"
                          type="radio"
                          value="POSITIVE"
                        />
                        <label
                          class="usa-radio__label"
                          for="3-val-POSITIVE"
                        >
                          Positive (+)
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="3-val-NEGATIVE"
                          name="covid-test-result-abc"
                          type="radio"
                          value="NEGATIVE"
                        />
                        <label
                          class="usa-radio__label"
                          for="3-val-NEGATIVE"
                        >
                          Negative (-)
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="3-val-UNDETERMINED"
                          name="covid-test-result-abc"
                          type="radio"
                          value="UNDETERMINED"
                        />
                        <label
                          class="usa-radio__label"
                          for="3-val-UNDETERMINED"
                        >
                          Inconclusive
                        </label>
                      </div>
                    </div>
                  </fieldset>
                </div>
                <div
                  class="prime-test-result-submit"
                >
                  <button
                    class="usa-button usa-button--outline usa-button-disabled"
                    disabled=""
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <div
        class="position-relative grid-container prime-container prime-queue-item card-container"
        data-testid="test-card-def"
      >
        <div
          aria-hidden="true"
          class="sr-queue-item-submit-loader z-top position-absolute height-full width-full text-center radius-lg"
        >
          <h4
            class="margin-top-6 margin-bottom-5"
          >
            Submitting test data for 
            Smith, Jane
            ...
          </h4>
          <img
            alt="submitting"
            class="square-5"
            src="loader.svg"
          />
        </div>
        <div
          class="prime-card-container"
        >
          <button
            aria-label="Close"
            class="prime-close-button"
          >
            <span
              class="fa-layers"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-circle fa-w-16 fa-inverse fa-2x "
                data-icon="circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8z"
                  fill="currentColor"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times-circle fa-w-16 fa-2x "
                data-icon="times-circle"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <div
            class="grid-row"
          >
            <div
              class="tablet:grid-col-9"
            >
              <div
                class="grid-row prime-test-name usa-card__header"
                id="patient-name-header"
              >
                <div
                  class="card-header"
                >
                  <div
                    class="card-name"
                  >
                    Smith, Jane
                  </div>
                  <div
                    class="card-dob"
                  >
                    Date of birth:
                    <span
                      class="card-date"
                    >
                       
                      02/01/2021
                    </span>
                  </div>
                </div>
                <button
                  class="timer-button timer-reset"
                  data-testid="timer"
                >
                  <span>
                    10:00
                  </span>
                   
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-stopwatch fa-w-14 "
                    data-icon="stopwatch"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 448 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M432 304c0 114.9-93.1 208-208 208S16 418.9 16 304c0-104 76.3-190.2 176-205.5V64h-28c-6.6 0-12-5.4-12-12V12c0-6.6 5.4-12 12-12h120c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-28v34.5c37.5 5.8 71.7 21.6 99.7 44.6l27.5-27.5c4.7-4.7 12.3-4.7 17 0l28.3 28.3c4.7 4.7 4.7 12.3 0 17l-29.4 29.4-.6.6C419.7 223.3 432 262.2 432 304zm-176 36V188.5c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12V340c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="margin-top-2 margin-left-2 margin-bottom-2"
              >
                <div
                  class="grid-row"
                >
                  <div
                    class="grid-col-4 flex-col-container padding-right-2"
                  >
                    <button
                      class="usa-button usa-button--unstyled test-questionnaire-btn"
                      type="button"
                    >
                      Test questionnaire
                    </button>
                    <div
                      class="margin-bottom-1"
                    >
                      <span
                        class="usa-tag"
                      >
                        PENDING
                      </span>
                    </div>
                  </div>
                  <div
                    class="flex-col-container"
                  >
                    <div
                      class=""
                    >
                      Test date and time
                    </div>
                    <div
                      class="test-date-time-container"
                    >
                      <input
                        class="card-test-input"
                        data-testid="test-date"
                        hidden=""
                        id="test-date"
                        max="2021-08-01"
                        min="2020-01-01"
                        name="test-date"
                        type="date"
                        value="2021-08-01"
                      />
                      <input
                        class="card-test-input"
                        data-testid="test-time"
                        hidden=""
                        name="test-time"
                        step="60"
                        type="time"
                        value="08:20"
                      />
                      <div
                        class="check-box-container"
                      >
                        <div
                          class="usa-checkbox"
                        >
                          <input
                            checked=""
                            class="usa-checkbox__input margin"
                            id="current-date-check-def"
                            type="checkbox"
                            value="true"
                          />
                          <label
                            class="usa-checkbox__label margin-0 margin-right-05em"
                            for="current-date-check-def"
                          >
                            Current date/time
                          </label>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="queue-item__form prime-ul grid-row"
                >
                  <div
                    class="prime-li flex-align-self-end tablet:grid-col-4 padding-right-2"
                  >
                    <div
                      class="usa-form-group prime-dropdown  card-dropdown"
                    >
                      <label
                        class="usa-label"
                        for="4"
                      >
                        Device
                      </label>
                      <select
                        aria-required="false"
                        class="usa-select"
                        id="4"
                        name="testDevice"
                      >
                        <option
                          value="f5c7658d-a0d5-4ec5-a1c9-eafc85fe7554"
                        >
                          LumiraDx
                        </option>
                        <option
                          value="0f3d7426-3476-4800-97e7-3de8a93b090c"
                        >
                          Quidel Sofia 2
                        </option>
                      </select>
                    </div>
                  </div>
                  <div
                    class="prime-li flex-align-self-end tablet:grid-col-5 padding-right-2"
                  >
                    <div
                      class="usa-form-group prime-dropdown  card-dropdown"
                    >
                      <label
                        class="usa-label"
                        for="5"
                      >
                        Swab type
                      </label>
                      <select
                        aria-required="false"
                        class="usa-select"
                        id="5"
                        name="swabType"
                      >
                        <option
                          value="6e4ccb35-d177-4ea0-9226-653358f1e081"
                        >
                          Nasopharyngeal swab
                        </option>
                        <option
                          value="6aa957dc-add2-4938-8788-935aec3276d4"
                        >
                          Swab of internal nose
                        </option>
                      </select>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="tablet:grid-col-3 prime-test-result"
            >
              <form
                class="usa-form"
              >
                <h4
                  class="prime-radio__title"
                >
                  SARS-CoV-2 results
                </h4>
                <div
                  class="usa-form-group prime-radio__group"
                >
                  <fieldset
                    class="usa-fieldset prime-radios"
                  >
                    <legend
                      class="usa-legend usa-sr-only"
                    >
                      Test result
                    </legend>
                    <div
                      class=""
                    >
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="6-val-POSITIVE"
                          name="covid-test-result-def"
                          type="radio"
                          value="POSITIVE"
                        />
                        <label
                          class="usa-radio__label"
                          for="6-val-POSITIVE"
                        >
                          Positive (+)
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="6-val-NEGATIVE"
                          name="covid-test-result-def"
                          type="radio"
                          value="NEGATIVE"
                        />
                        <label
                          class="usa-radio__label"
                          for="6-val-NEGATIVE"
                        >
                          Negative (-)
                        </label>
                      </div>
                      <div
                        class="usa-radio"
                      >
                        <input
                          class="usa-radio__input"
                          data-required="false"
                          id="6-val-UNDETERMINED"
                          name="covid-test-result-def"
                          type="radio"
                          value="UNDETERMINED"
                        />
                        <label
                          class="usa-radio__label"
                          for="6-val-UNDETERMINED"
                        >
                          Inconclusive
                        </label>
                      </div>
                    </div>
                  </fieldset>
                </div>
                <div
                  class="prime-test-result-submit"
                >
                  <button
                    class="usa-button usa-button--outline usa-button-disabled"
                    disabled=""
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
